test/core/utils/filter-html-attrs.js

```diff
@@ -0,0 +1,246 @@
+describe('axe.utils.filterHtmlAttrs', function() {
+  'use strict';
+  var filterHtmlAttrs = axe.utils.filterHtmlAttrs;
+  var html = '<label> My Label <input type="text" value="my value" /></label>';
+  var expected = '<label> My Label <input value="my value" /></label>';
+
+  it('should return string if no attributes are passed', function() {
+    assert.equal(filterHtmlAttrs(html), html);
+  });
+
+  it('should filter attribute if exists', function() {
+    assert.equal(filterHtmlAttrs(html, { type: true }), expected);
+  });
+
+  it('should filter attribute if matches value', function() {
+    assert.equal(filterHtmlAttrs(html, { type: 'text' }), expected);
+  });
+
+  it('should not filter attribute if does not match value', function() {
+    assert.equal(filterHtmlAttrs(html, { type: 'submit' }), html);
+  });
+
+  it('should filter attribute if matches list of values', function() {
+    assert.equal(filterHtmlAttrs(html, { type: ['submit', 'text'] }), expected);
+  });
+
+  it('should not filter attribute if does not match list of values', function() {
+    assert.equal(filterHtmlAttrs(html, { type: ['submit', 'password'] }), html);
+  });
+
+  it('should filter attribute if node matches nodeName matcher', function() {
+    assert.equal(
+      filterHtmlAttrs(html, {
+        type: {
+          nodeName: 'input'
+        }
+      }),
+      expected
+    );
+  });
+
+  it('should filter attribute if node matches attributes matcher (boolean)', function() {
+    assert.equal(
+      filterHtmlAttrs(html, {
+        type: {
+          attributes: {
+            value: true
+          }
+        }
+      }),
```

Comment:
This definitely looks odd. You're doing an "and" relationship of two equal things by nesting. That seems pretty counter-intuitive. Is there any reason we can't use a simplified version of the matcher we already have? So it would be this:

```suggestion
      filterHtmlAttrs(html, {
        attributes: {
            type: true,
            value: true
          }
        }
      }),
```

I realise that makes some of these things a little more verbose, but I definitely think there's a thing to be said here for not introducing another matcher format to axe-core. Even if this implementation is a subset of the other one.